applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -198,6 +198,69 @@ void SymbolicStokes<TElementData>::AddBoundaryTraction(
     noalias(rRHS) += prod(rData.lhs,values);
 }
 
+template <class TElementData>
+void SymbolicStokes<TElementData>::Calculate(
+    const Variable<double> &rVariable,
+    double &rOutput,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    rOutput = 0.0;
+
+    if (rVariable == HEAT_FLUX) {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+        rOutput = 0.0;
+        // Shape functions
+        const auto &r_geometry = this->GetGeometry();
+
+        const unsigned int num_nodes = r_geometry.PointsNumber();
+        Vector data_N(num_nodes);
+        for (unsigned int i = 0; i < num_nodes; i++) {
+            data_N[i] = 1.0 / num_nodes;
+        }
+        data.N = data_N;
+        // Shape functions gradients
+        const GeometryData::IntegrationMethod integration_method = GeometryData::GI_GAUSS_2;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+        Vector DetJ;
+        r_geometry.ShapeFunctionsIntegrationPointsGradients(shape_derivatives, DetJ, integration_method);
+        data.DN_DX = shape_derivatives[0]; // Note: Valid only for linear elements

```

Comment:
I think you can use the `GeometryUtils` in here since sizes are known at compile time. These are much more efficient as they use bounded arrays.

Reply:
Forget about my, comment, since as far as I remember you were instantiating this element for wedges right?

Reply:
Yes, that is correct. All this mumbo jumbo because of wedges..

Reply:
OK.. we'll need to live with that until we extend the `GeometryUtils`...